app/models/friendship.rb

```diff
@@ -0,0 +1,14 @@
+class Friendship < ApplicationRecord
+  # after_create_commit :create_inverse_frienship
+  # after_destroy_commit :destroy_inverse_frienship
+  before_create :order_params
+
+  belongs_to :user
+  belongs_to :friend, class_name: 'User'
+
+  private
+
+  def order_params
```

Comment:
Awesome method you created here! :) I've seen other implementations of enforcing record uniqueness but so far this is one of the most clever!